### PR TITLE
Moved code to convert ApiResult -> TrainTimes to extension function...

### PR DESCRIPTION
**Pull request by abeshi-softwire**
... in common.kt

### CODE REVIEW

File: SharedCode/src/commonMain/kotlin/common.kt [hunk 1]
```diff
@@ -51,4 +51,20 @@ fun getEpochFromUTC(s: String): Long {
...
+}
+
+fun ApiResult.toTrainTimes(): TrainTimes{
+    val journeys = mutableListOf<TrainTimes.Journey>()
+    this.outboundJourneys.forEach { journey ->
+        journeys.plusAssign(TrainTimes.Journey(
+            getPrice(journey.tickets),
+            getEpochFromUTC(journey.departureTime),
+            getEpochFromUTC(journey.arrivalTime),
+            journey.legs.size-1
```
Comment (thevirtuoso1973) [inline SharedCode/src/commonMain/kotlin/common.kt:70]: I noticed a small discrepancy with journey.legs.size and journey.tickets.size which I will investigate now before I merge.

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:70]: That's interesting, let me know if you want me to take a look or anything

Comment (thevirtuoso1973) [reply SharedCode/src/commonMain/kotlin/common.kt:70]: Okay it is not the case as previously thought that `(number legs) - 1 = (number tickets) - 1`. But this problem is not related to the purpose of this PR so I shall merge.

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:70]: You're right, one ticket can do multiple legs

**Pull request merged**